File: src/main/java/dev/haarolean/gnutardbot/app/props/BotProperties.java
package dev.haarolean.gnutardbot.app.props;

import jakarta.validation.constraints.NotEmpty;
import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.validation.annotation.Validated;

import java.util.Set;

@ConfigurationProperties("bot")
@Configuration
@Validated
@Data
public class BotProperties {

    @NotEmpty
    private String token;
    private String chatId;
    private Set<Long> admins;

}

File: src/main/kotlin/dev/haarolean/gnutardbot/app/props/BotProperties.kt
package dev.haarolean.gnutardbot.app.props

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("bot")
//@Configuration
//@Validated
data class BotProperties(
     val token:  String,
     val chatId: String,
     val admins: Set<Long>
)
